File: Ruby/TestP1.rb
#encoding utf-8

require_relative 'Directions'
require_relative 'Dice'
require_relative 'GameCharacter'
require_relative 'Orientation'
require_relative 'Shield'
require_relative 'Weapon'
require_relative 'GameState'

class TestP1
    def main
        puts "Probando la practica 1:\n"

        puts "---------------Enum Directions-------------------"
        puts Irrgarten::Directions::LEFT
        puts Irrgarten::Directions::RIGHT
        puts Irrgarten::Directions::UP
        puts Irrgarten::Directions::DOWN

        puts "\n--------------Enum GameCharacter------------------"
        puts Irrgarten::GameCharacter::PLAYER
        puts Irrgarten::GameCharacter::MONSTER

        puts "\n--------------Enum Orientation-----------------"
        puts Irrgarten::Orientation::VERTICAL
        puts Irrgarten::Orientation::HORIZONTAL

        puts "\n--------------Creando instancias de clase-----------------"
        arma1 = Irrgarten::Weapon.new(3.2, 4)
        arma2 = Irrgarten::Weapon.new(1.1, 2)
        shield1 = Irrgarten::Shield.new(2.3, 3)
        shield2 = Irrgarten::Shield.new(1.2, 1)
        juego = Irrgarten::GameState.new("Garden of Bambam", "Mario:vivo, Miguel:vivo, Andrea:muerto", "Minotauro, Manticora",0, false,"Se ha creado el tablero de juego")

        puts "\n--------------Muestra de armas y escudos-----------------"
        puts "Arma 1: " + arma1.to_s
        puts "Arma 2: " + arma2.to_s
        puts "Escudo 1: " + shield1.to_s
        puts "Escudo 2: " + shield2.to_s
        
        puts "\nAtacando con arma 1, poder: " + arma1.attack.to_s
        puts "Compruebo usos: " + arma1.to_s
        puts "¿Tras el ataque se descarta? " + arma1.discard.to_s
        puts "\nProtegiendo con shield1, proteccion: " + shield1.protect.to_s
        puts "Compruebo usos: " + shield1.to_s
        puts "¿Tras proteger se descarta? " + shield1.discard.to_s

        puts "\n--------------Estado actual del juego-----------------"
        puts "labyrinth: " + juego.get_labyrinthv
        puts "players: " + juego.get_players
        puts "monsters: " + juego.get_monsters
        puts "currentplayer: " + juego.get_current_player.to_s
        puts "winner: " + juego.get_winner.to_s
        puts "log: " + juego.get_log

        puts "\n----------------Prueba del dado-------------------"
        25.times do
            puts "\nRandom Position: #{Irrgarten::Dice.random_pos(100)}"
            puts "Who Starts: #{Irrgarten::Dice.who_starts(10)}"
            puts "Random Intelligence: #{Irrgarten::Dice.randomIntelligence()}"
            puts "Random Strength: #{Irrgarten::Dice.randomStrenght()}"
            puts "Resurrect Player: #{Irrgarten::Dice.resurrectPlayer()}"
            puts "Weapons Reward: #{Irrgarten::Dice.weaponsReward()}"
            puts "Shields Reward: #{Irrgarten::Dice.shieldsReward()}"
            puts "Health Reward: #{Irrgarten::Dice.healthReward()}"
            puts "Weapon Power: #{Irrgarten::Dice.weaponPower()}"
            puts "Shield Power: #{Irrgarten::Dice.shieldPower()}"
            puts "Numero de usos restante: #{Irrgarten::Dice.usesLeft()}"
            puts "Intensidad: #{Irrgarten::Dice.intensity(5.0)}"
        end
        
    end
end

test = TestP1.new
test.main

File: Java/Irrgarten/src/irrgarten/TestP1.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author mariorl
 */
public class TestP1 {
    public static void main(String[] args) {
        System.out.println("Probando la practica 1:\n");
        
        System.out.println("---------------Enum Directions-------------------");
        System.out.println(Directions.DOWN);
        System.out.println(Directions.UP);
        System.out.println(Directions.LEFT);
        System.out.println(Directions.RIGHT);
    
        System.out.println("\n--------------Enum Orientation------------------");
        System.out.println(Orientation.HORIZONTAL);
        System.out.println(Orientation.VERTICAL);
        
        System.out.println("\n--------------Enum GameCharacter-----------------");
        System.out.println(GameCharacter.PLAYER);
        System.out.println(GameCharacter.MONSTER);
        
        System.out.println("\n--------------Creando instancias de clase-------------");
        Weapon arma1 = new Weapon(2.3f, 4);
        Weapon arma2 = new Weapon(1.7f, 1);
        Shield shield1 = new Shield(2.9f, 5);
        Shield shield2 = new Shield(4.1f, 2);
        GameState juego = new GameState("Garden of Bambam", "Mario:vivo, Miguel:vivo, Andrea:muerto", "Minotauro, Manticora",0, false,"Se ha creado el tablero de juego");
        
        System.out.println("\n--------------Muestra de armas y escudos----------------");
        System.out.println("arma1: " + arma1.toString());
        System.out.println("arma2: " + arma2.toString());
        System.out.println("shield1: " + shield1.toString());
        System.out.println("shield2: " + shield2.toString());
        
        System.out.println("\nAtaco con arma1: " + arma1.attack());
        System.out.println("arma1: " + arma1.toString());
        System.out.println("¿Tras el ataque se descarta?: " + arma1.discard());
        System.out.println("\nProtejo con shield1: " + shield1.protect());
        System.out.println("shield1: " + shield1.toString());
        System.out.println("¿Tras proteger se descarta?: " + shield1.discard());
        System.out.println("\nIntento de descartar arma2: " + arma2.discard());
        System.out.println("Intento de descartar shield2: " + arma2.discard());
        
        
        System.out.println("\n---------------Estado actual del juego------------------");
        System.out.println("labyrinth: " + juego.getLabyrinth());
        System.out.println("players: " + juego.getPlayers());
        System.out.println("monsters: " + juego.getMonsters());
        System.out.println("currentPlayer: " + juego.getCurrentPlayer());
        System.out.println("ganador: " + juego.getWinner());
        System.out.println("log: " + juego.getLog());
        
        System.out.println("\n---------------Prueba del dado---------------");
        for(int i = 0 ; i < 3 ; i++) {
            System.out.println("Posicion: " + Dice.randomPos(100));
            System.out.println("Inicia jugador: " + Dice.whoStarts(10));
            System.out.println("Con nivel de inteligencia: " + Dice.randomIntelligence());
            System.out.println("Con nivel de fuerza: " + Dice.randomStrenght());
            System.out.println("¿Resucita?: " + Dice.resurrectPlayer());
            System.out.println("Recompensa de armas: " + Dice.weaponsReward());
            System.out.println("Recompensa de escudos: " + Dice.shieldsReward());
            System.out.println("Recompensa de vida: " + Dice.healthReward());
            System.out.println("Poder del arma: " + Dice.weaponPower());
            System.out.println("Poder del escudo: " + Dice.shieldPower());
            System.out.println("Numero de usos restante: " + Dice.usesLeft());
            System.out.println("Intensidad: " + Dice.intensity(5.0f));
            System.out.println("\n");
        }
        
    }
}
